Client/Coordinators/Router/Route.swift

```diff
@@ -46,26 +66,175 @@ enum Route {
+                self = .fxaSignIn(FxALaunchParams(entrypoint: .fxaDeepLinkNavigation, query: url.getQuery()))
+
+            case .openUrl:
+                self = .search(url: urlQuery, isPrivate: isPrivate)
+
+            case .openText:
+                self = .search(query: urlScanner.value(query: "text") ?? "")
+
+            case .glean:
+                self = .glean(url: url)
+
+            case .widgetMediumTopSitesOpenUrl:
+                // Widget Top sites - open url
+                self = .search(url: urlQuery, isPrivate: isPrivate)
+
+            case .widgetSmallQuickLinkOpenUrl:
+                // Widget Quick links - small - open url private or regular
+                self = .search(url: urlQuery, isPrivate: isPrivate)
+
+            case .widgetMediumQuickLinkOpenUrl:
+                // Widget Quick Actions - medium - open url private or regular
+                self = .search(url: urlQuery, isPrivate: isPrivate)
+
+            case .widgetSmallQuickLinkOpenCopied, .widgetMediumQuickLinkOpenCopied:
+                // Widget Quick links - medium - open copied url
+                if !UIPasteboard.general.hasURLs {
+                    let searchText = UIPasteboard.general.string ?? ""
+                    self = .search(query: searchText)
+                } else {
+                    let url = UIPasteboard.general.url
+                    let isPrivate = UserDefaults.standard.bool(forKey: PrefsKeys.LastSessionWasPrivate)
+                    self = .search(url: url, isPrivate: isPrivate)
+                }
+
+            case .widgetSmallQuickLinkClosePrivateTabs, .widgetMediumQuickLinkClosePrivateTabs:
+                // Widget Quick links - medium - close private tabs
+                self = .action(action: .closePrivateTabs)
+
+            case .widgetTabsMediumOpenUrl:
+                // Widget Tabs Quick View - medium
+                let tabs = SimpleTab.getSimpleTabs()
+                if let uuid = urlScanner.value(query: "uuid"), !tabs.isEmpty {
+                    let tab = tabs[uuid]
+                    self = .search(url: tab?.url, tabId: uuid)
+                } else {
+                    self = .search(url: nil, isPrivate: false)
+                }
+
+            case .widgetTabsLargeOpenUrl:
+                // Widget Tabs Quick View - large
+                let tabs = SimpleTab.getSimpleTabs()
+                if let uuid = urlScanner.value(query: "uuid"), !tabs.isEmpty {
+                    let tab = tabs[uuid]
+                    self = .search(url: tab?.url, tabId: uuid)
+                } else {
+                    self = .search(url: nil, isPrivate: false)
+                }
+
+            case .fxaSignIn:
+                return nil
+            }
+
+            recordTelemetry(input: host, isPrivate: isPrivate)
+        } else if urlScanner.isHTTPScheme {
+            TelemetryWrapper.gleanRecordEvent(category: .action, method: .open, object: .asDefaultBrowser)
```

Comment:
Should we use `func recordTelemetry(input: DeeplinkInput.Host,` for this case too?
```suggestion
            TelemetryWrapper.gleanRecordEvent(category: .action, method: .open, object: .asDefaultBrowser)
```

Reply:
I don't have a `host` case for this, but I can take a look to see if I can integrate it.